src/Testing/CodeHacking/CodeHacker.php

```diff
@@ -0,0 +1,528 @@
+	 */
+	public function stream_cast( $cast_as ) {
+		return $this->handle;
+	}
+
+	/**
+	 * Close a resource.
+	 */
+	public function stream_close() {
+		fclose( $this->handle );
+	}
+
+	/**
+	 * Tests for end-of-file on a file pointer.
+	 *
+	 * @return bool TRUE if the read/write position is at the end of the stream and if no more data is available to be read, or FALSE otherwise.
+	 */
+	public function stream_eof() {
+		return feof( $this->handle );
+	}
+
+	/**
+	 * Flushes the output.
+	 *
+	 * @return bool TRUE if the cached data was successfully stored (or if there was no data to store), or FALSE if the data could not be stored.
+	 */
+	public function stream_flush() {
+		return fflush( $this->handle );
+	}
+
+	/**
+	 * Advisory file locking.
+	 *
+	 * @param int $operation LOCK_SH, LOCK_EX, LOCK_UN, or LOCK_NB.
+	 *
+	 * @return bool TRUE on success or FALSE on failure.
+	 */
+	public function stream_lock( $operation ) {
+		return $operation
+			? flock( $this->handle, $operation )
+			: true;
+	}
+
+	/**
+	 * Change stream metadata.
+	 *
+	 * @param string $path The file path or URL to set metadata. Note that in the case of a URL, it must be a :// delimited URL. Other URL forms are not supported.
+	 * @param int    $option STREAM_META_TOUCH, STREAM_META_OWNER_NAME, STREAM_META_OWNER, STREAM_META_GROUP_NAME, STREAM_META_GROUP, or STREAM_META_ACCESS.
+	 * @param mixed  $value Depends on $option.
+	 *
+	 * @return bool TRUE on success or FALSE on failure. If option is not implemented, FALSE should be returned.
+	 */
+	public function stream_metadata( $path, $option, $value ) {
+		switch ( $option ) {
+			case STREAM_META_TOUCH:
+				$value += array( null, null );
+				return $this->native( 'touch', $path, $value[0], $value[1] );
+			case STREAM_META_OWNER_NAME:
+			case STREAM_META_OWNER:
+				return $this->native( 'chown', $path, $value );
+			case STREAM_META_GROUP_NAME:
+			case STREAM_META_GROUP:
+				return $this->native( 'chgrp', $path, $value );
+			case STREAM_META_ACCESS:
+				return $this->native( 'chmod', $path, $value );
+		}
+	}
+
+	/**
+	 *  Opens file or URL. Note that this is where the hacking actually happens.
+	 *
+	 * @param string $path Specifies the URL that was passed to the original function.
+	 * @param string $mode The mode used to open the file, as detailed for fopen().
+	 * @param int    $options Holds additional flags set by the streams API: STREAM_USE_PATH, STREAM_REPORT_ERRORS.
+	 * @param string $opened_path If the path is opened successfully, and STREAM_USE_PATH is set in options, opened_path should be set to the full path of the file/resource that was actually opened.
+	 *
+	 * @return bool TRUE on success or FALSE on failure.
+	 */
+	public function stream_open( $path, $mode, $options, &$opened_path ) {
+		$use_path = (bool) ( $options & STREAM_USE_PATH );
+		if ( 'rb' === $mode && self::path_in_white_list( $path ) && 'php' === pathinfo( $path, PATHINFO_EXTENSION ) ) {
```

Comment:
What do you mean by other modes? Directories perhaps?

Reply:
I was thinking about protocols but based on other comments that do not make sense here because we will never use anything else than `file`. resolving this comment.